bazel/grpc_build_system.bzl

```diff
@@ -389,14 +450,55 @@ def grpc_generate_one_off_targets():
 def grpc_generate_objc_one_off_targets():
     pass
 
-def grpc_sh_test(name, srcs, args = [], data = [], tags = []):
-    native.sh_test(
-        name = name,
-        srcs = srcs,
-        args = args,
-        data = data,
-        tags = tags,
-    )
+def grpc_sh_test(name, srcs = [], args = [], data = [], uses_polling = True, size = "medium", timeout = None, tags = [], exec_compatible_with = [], exec_properties = {}, shard_count = None, flaky = None, exclude_pollers = [], test_each_engine = True):
```

Comment:
why do you need to change grpc_sh_test definition? It seems that all the test that use pollers/event engine are grpc_cc_tests?

Reply:
You missed its usage in `test/core/end2end/generate_tests.bzl`, PTAL. The sh_tests need to be parameterized for the product of pollers and engines, same as cc_test.